toolchain/mfc/bench.py

```diff
@@ -116,6 +117,12 @@ def _lock_to_str(lock):
             if target.name not in lhs_summary or target.name not in rhs_summary:
                 continue
 
+            if float(f"{lhs_summary[target.name]}") <= 0:
```

Comment:
https://stackoverflow.com/a/23639915/8115551

Reply:
This post details microsecond differences in runtimes. This if statement is reached 12 times total, and will never be reached more than order 10 times.

Reply:
i think the point is that if the output of `lhs_summary` is a floating point number (has a decimal) then what you did here might not work? i may be missing the point from @henryleberre, though. you are also comparing a float against an integer, not sure how that works.

Reply:
@wilfonba is testing on this line if a number is valid (exists, is not NaN, and is greater than 0). Take a look at this test:
```
~ $ python3
Python 3.12.3 (main, Apr  9 2024, 08:09:14) [Clang 15.0.0 (clang-1500.3.9.4)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> float("")
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ValueError: could not convert string to float: ''
>>> float("NaN")
nan
>>> float("NaN") <= 0
False
>>>
```
This shows a few issues with the current approach:
- In some cases, calling `float()` will produce an exception we don't capture. The program will terminate with an undesirable traceback that does not explain the error.
- In Python `nan <= 0` is `False` so this value passes the check.
- There are probably more examples..

This is why I pointed out a resource to properly check if a string is a _good_ number.

Reply:
I deleted these statements entirely in the latest commit because they were checking the wrong thing.